src/containerapp/azext_containerapp/_client_factory.py

```diff
@@ -0,0 +1,74 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands.client_factory import get_mgmt_service_client
+from azure.cli.core.profiles import ResourceType
+
+from knack.util import CLIError
+
+
+# pylint: disable=inconsistent-return-statements
+def ex_handler_factory(no_throw=False):
+    def _polish_bad_errors(ex):
+        import json
+        from knack.util import CLIError
+        try:
+            content = json.loads(ex.response.content)
+            if 'message' in content:
+                detail = content['message']
+            elif 'Message' in content:
+                detail = content['Message']
+
+            ex = CLIError(detail)
+        except Exception:  # pylint: disable=broad-except
+            pass
+        if no_throw:
+            return ex
+        raise ex
+    return _polish_bad_errors
+
+
+def handle_raw_exception(e):
+    import json
+
+    stringErr = str(e)
+
+    if "{" in stringErr and "}" in stringErr:
+        jsonError = stringErr[stringErr.index("{"):stringErr.rindex("}") + 1]
+        jsonError = json.loads(jsonError)
+
+        if 'error' in jsonError:
+            jsonError = jsonError['error']
+
+            if 'code' in jsonError and 'message' in jsonError:
+                code = jsonError['code']
+                message = jsonError['message']
+                raise CLIError('({}) {}'.format(code, message))
+        elif "Message" in jsonError:
+            message = jsonError["Message"]
+            raise CLIError(message)
+        elif "message" in jsonError:
+            message = jsonError["message"]
+            raise CLIError(message)
```

Comment:
Please use the specific error type from `azure.cli.core.azclierror` instead of `CLIError`

Reply:
Removed `CLIError` to use more specific error types